### PR TITLE
SplunkPy - add Python requests handler

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/24269

## Description
Added Python requests handler, which handles HTTP requests better than `httplib.HTTPSConnection` which Splunk SDK uses.

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Documentation

### CODE REVIEW

**Comment by Itay4** [comment]
@glicht please have a look at my latest commits
added support for proxy using requests handler

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml [hunk 1]
```diff
@@ -101,6 +101,10 @@ configuration:
   name: fetch_time
   required: false
   type: 0
+- display: Use Python requests handler
+  name: use_requests_handler
+  required: false
```
Comment (anara123) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:106]: don't we want it to be the requests handler by default?

**Comment by kirbles19** [comment]
@Itay4 Done.

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml (hunk 1 above)
Comment (Itay4) [reply Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.yml:106]: the splunk sdk is pretty fragile in regards to connection, and im concerned making this by default might break bc
i suggest we gain some experience with it and then maybe in set it to be by default

**Pull request merged**